### PR TITLE
Reduce core->block in cmd_print.c

### PR DESCRIPTION
**Pull request by MaskRay**

### CODE REVIEW

File: libr/core/cmd_print.c [hunk 1]
```diff
@@ -5536,7 +5536,7 @@ static int cmd_print(void *data, const char *input) {
 		}
 		break;
 	case 'f': // "pf"
-		cmd_print_format (core, input, len);
```
Comment (radare) [inline libr/core/cmd_print.c]: Maybe len must go after block

File: libr/core/cmd_print.c [hunk 2]
```diff
@@ -4467,20 +4467,20 @@ static int cmd_print(void *data, const char *input) {
...
 				}
-				if (block) {
+				if (block1) {
 					if (*input == 'D') { // pD
-						free (block);
-						block = malloc (l);
-						r_core_read_at (core, addr - l, block, l); // core->blocksize);
-						core->num->value = r_core_print_disasm (core->print, core, addr - l, block, l, l, 0, 1, formatted_json);
+						free (block1);
+						block1 = malloc (l);
```
Comment (radare) [inline libr/core/cmd_print.c]: Null deref if malloc fails

File: libr/core/cmd_print.c [hunk 3]
```diff
@@ -4467,20 +4467,20 @@ static int cmd_print(void *data, const char *input) {
 		}
 		if (!processed_cmd) {
 			ut64 addr = core->offset;
-			ut8 *block = NULL;
+			ut8 *block1 = NULL;
 			ut64 start;
 
 			if (bw_disassemble) {
-				block = malloc (core->blocksize);
+				block1 = malloc (core->blocksize);
```
Comment (radare) [inline libr/core/cmd_print.c]: Nullchk

Comment (radare) [reply libr/core/cmd_print.c]: Nvm its there

File: libr/core/cmd_print.c [hunk 4]
```diff
@@ -4038,9 +4038,9 @@ static int cmd_print(void *data, const char *input) {
 		case 'j': // "pIj" is the same as pDj
 			if (l != 0) {
 				if (input[2]) {
-					cmd_pDj (core, input + 2);
+					cmd_pDj (core, input + 2, block);
```
Comment (radare) [inline libr/core/cmd_print.c]: Imho cmd_pDj should not get block as argument, and if it does it should be block and lentgh, not user input

File: libr/core/cmd_print.c (hunk 1 above)
Comment (MaskRay) [reply libr/core/cmd_print.c]: Done

**Pull request merged**